processing/src/test/java/org/apache/druid/collections/spatial/ImmutableRTreeTest.java

```diff
@@ -702,4 +703,37 @@ public void testPreciseRadiusBoundFilter()
     ImmutableBitmap finalSet = bf.union(points);
     Assert.assertTrue(finalSet.size() == 100);
   }
+
+  @Test
+  public void testForImmutableRTreeImmutability()
+  {
+    BitmapFactory bf = new RoaringBitmapFactory();
+    RTree tree = new RTree(2, new LinearGutmanSplitStrategy(0, 50, bf), bf);
+    float centerLat = 37.4133961f;
+    float centerLong = -122.1224665f;
+    float[][] insidePoints = SpatialUtils.generateGeoCoordinatesAroundCircle(centerLat, centerLong, 100, 100, true);
+    for (int i = 0; i < insidePoints.length; i++) {
+      tree.insert(insidePoints[i], i);
+    }
+    float[][] outsidePoints = SpatialUtils.generateGeoCoordinatesAroundCircle(centerLat, centerLong, 100, 100, false);
+    for (int i = 0; i < outsidePoints.length; i++) {
+      tree.insert(outsidePoints[i], i);
+    }
+    final ImmutableRTree searchTree = ImmutableRTree.newImmutableFromMutable(tree);
+    ByteBuffer buffer = ByteBuffer.wrap(searchTree.toBytes());
+    ImmutableRTreeObjectStrategy objectStrategy = new ImmutableRTreeObjectStrategy(bf);
+    ImmutableRTree builtTree = objectStrategy.fromByteBuffer(buffer, searchTree.toBytes().length);
+    buffer.position(10);
```

Comment:
Should this be called before calling the `fromByteBuffer` method?

Reply:
nope, it will not be able to create object if we call it before. This make sure that subsequent buffer mutation does not affect  previously created objects.

Reply:
Thanks for the clarification!